hera_cal/lst_stack/calibration.py

```diff
@@ -132,240 +174,350 @@ def lstbin_absolute_calibration(
-                if calibrate_inplace:
-                    auto_stack.data[:, apidx, :, polidx] /= bl_gain
+    # Check to see if auto_model is provided if use_autos_for_abscal is True
+    if auto_stack is not None and use_autos_for_abscal and auto_model is None:
+        raise ValueError(
+            "auto_model must be provided if auto_stack is provided and use_autos_for_abscal is True"
+        )
+
+    # Dictionary for storing calibration parameters
+    calibration_parameters = {}
+
+    # Check to see if all nights are flagged
+    all_nights_flagged = np.all(stack.flags)
+
+    # Get the unique antenna-polarizations
+    unique_pols = list(
+        set(sum(map(list, [utils.split_pol(pol) for pol in stack.pols]), []))
+    )
+
+    # Perform amplitude calibration
+    if run_amplitude_cal and not all_nights_flagged:
+        # Store gain solutions in paramter dictionary
+        amp_cal_params = _lstbin_amplitude_calibration(
+            stack=stack,
+            model=model,
+            auto_stack=auto_stack,
+            auto_model=auto_model,
+            use_autos_for_abscal=use_autos_for_abscal,
+        )
+        for key in amp_cal_params:
+            calibration_parameters[key] = amp_cal_params[key]
+
+    else:
+        # Fill in amplitude w/ ones if not running amplitude calibration
+        for pol in unique_pols:
+            calibration_parameters[f"A_{pol}"] = np.ones(
+                (len(stack.nights), stack.freq_array.size), dtype=complex
+            )
+
+    if run_phase_cal and not all_nights_flagged:
+        # Perform phase calibration
+        phs_cal_params, phase_gains = _lstbin_phase_calibration(
+            stack=stack,
+            model=model,
+            all_reds=all_reds,
+        )
+        for key in phs_cal_params:
+            calibration_parameters[key] = phs_cal_params[key]
+
+    else:
+        # Fill in phase w/ ones if not running phase calibration
+        # Tip-tilt gains are zeros with dimensions (N_nights by N_freqs by Ndims)
+        phase_gains = {}
+
+        for pol in unique_pols:
+            calibration_parameters[f"T_{pol}"] = np.zeros(
+                (len(stack.nights), stack.freq_array.size, 2), dtype=complex
+            )
+
+    # Compute antenna gains from calibration parameters and smooth
+    gains = _expand_degeneracies_to_ant_gains(
+        stack=stack,
+        amplitude_parameters=calibration_parameters,
+        phase_gains=phase_gains,
+        inpaint_bands=inpaint_bands,
+        auto_stack=auto_stack,
+        smooth_gains=smooth_gains,
+        smoothing_scale=smoothing_scale,
+        eigenval_cutoff=eigenval_cutoff,
+    )
+
+    # Iterate for each baseline in the stack and calibrate out gains
+    if calibrate_inplace:
+        for polidx, pol in enumerate(stack.pols):
+            for apidx, (ant1, ant2) in enumerate(stack.antpairs):
+                antpol1, antpol2 = utils.split_bl((ant1, ant2, pol))
+
+                # Compute gain and remove nans/infs
```

Comment:
I would hope that all nans/infs are flagged, but is it true?

Reply:
It should be true, I think I was slightly overdoing the removing nans/infs because they were bleeding into other channels when I was smoothing. If we're getting nans in places without flags, that 'sprobably indicative of other issues that we probably should just allow to fail so that we can fix them later. I removed it here.